packages/atomic/src/components/commerce/atomic-product-image/atomic-product-image.spec.ts

```diff
@@ -0,0 +1,405 @@
+
+    expect(productImage).toBeInTheDocument();
+    expect(productImage).toHaveAttribute('src', DEFAULT_IMAGE);
+  });
+
+  it('should render nothing when product is not available', async () => {
+    const {element, productImage} = await renderProductImage({
+      product: null as unknown as Product,
+    });
+
+    expect(element).toBeDefined();
+    expect(productImage).toBeNull();
+    expect(element).toHaveTextContent('');
+  });
+
+  describe('when product has single image', () => {
+    it('should render single image without carousel controls', async () => {
+      const {productImage, previousButton, nextButton, indicators} =
+        await renderProductImage();
+
+      expect(productImage).toBeTruthy();
+      expect(productImage).toBeInTheDocument();
+      expect(previousButton).not.toBeInTheDocument();
+      expect(nextButton).not.toBeInTheDocument();
+      expect(indicators).not.toBeInTheDocument();
+    });
+
+    it('should render default alt text when no imageAltField specified', async () => {
+      const {productImage} = await renderProductImage();
+
+      expect(productImage).toHaveAttribute(
+        'alt',
+        'Image 1 out of 1 for Test Product'
+      );
+    });
+  });
+  describe('when product has multiple images', () => {
+    let productImageObject: Awaited<ReturnType<typeof renderProductImage>>;
+
+    const IMAGES = [
+      'image1.jpg',
+      'image2.jpg',
+      'image3.jpg',
+      'image4.jpg',
+      'image5.jpg',
+    ];
+
+    beforeEach(async () => {
+      productImageObject = await renderProductImage({
+        product: buildFakeProduct({
+          ec_name: 'Test Product',
+          ec_thumbnails: IMAGES,
+        }),
+      });
+    });
+
+    it('should render carousel with navigation controls', () => {
+      const {productImage, previousButton, nextButton, indicators} =
+        productImageObject;
+      expect(productImage).toBeTruthy();
+      expect(previousButton).toBeTruthy();
+      expect(previousButton).toBeInTheDocument();
+      expect(nextButton).toBeTruthy();
+      expect(nextButton).toBeInTheDocument();
+      expect(indicators).toBeTruthy();
+      expect(indicators).toBeInTheDocument();
+    });
+
+    it('should display first image by default', () => {
+      const {productImage} = productImageObject;
+      expect(productImage).toHaveAttribute('src', IMAGES[0]);
+      expect(productImage).toHaveAttribute(
+        'alt',
+        `Image 1 out of ${IMAGES.length} for Test Product`
+      );
+    });
+
+    it('should render indicators for each image', () => {
+      const {indicators} = productImageObject;
+      expect(indicators?.children).toHaveLength(IMAGES.length);
+    });
+
+    it('should show active indicator for current image', () => {
+      const {activeIndicator, indicators} = productImageObject;
+      expect(indicators?.children[0]).toBe(activeIndicator);
+    });
+
+    describe('when clicking carousel controls', () => {
```

Comment:
This describe block is too "generic"

We should have three describe blocks instead:

- when clicking the previous button
- when clicking the next button
- when clicking an indicator

and group the related ` it`s under each. Otherwise, the test descriptions are not clear enough.